# US Medical Insurance cost #
    - analyze dataset on 
## Goals: ##
    - Find out the average age of the patients in the dataset.
    - Analyze where a majority of the individuals are from.
    - Look at the different costs between smokers vs. non-smokers.
    - Figure out what the average age is for someone who has at least one child in this dataset.
    - How does BMI (body mass index) differ between smokers and non-smokers in the dataset?
    - What is the average insurance charge for males versus females in the dataset?

In [ ]:
# Import necessary libraries
import pandas as pd

In [3]:
# Load the dataset from a CSV file
data = pd.read_csv('/CodeAcademy/Portfolio/Codecademy-Data-Science-Portfolio/data_files/insurance.csv')

In [4]:
# Print the loaded dataset to verify it was loaded successfully
print(data)

age     sex     bmi  children smoker     region      charges
0      19  female  27.900         0    yes  southwest  16884.92400
1      18    male  33.770         1     no  southeast   1725.55230
2      28    male  33.000         3     no  southeast   4449.46200
3      33    male  22.705         0     no  northwest  21984.47061
4      32    male  28.880         0     no  northwest   3866.85520
...   ...     ...     ...       ...    ...        ...          ...
1333   50    male  30.970         3     no  northwest  10600.54830
1334   18  female  31.920         0     no  northeast   2205.98080
1335   18  female  36.850         0     no  southeast   1629.83350
1336   21  female  25.800         0     no  southwest   2007.94500
1337   61  female  29.070         0    yes  northwest  29141.36030

[1338 rows x 7 columns]

In [9]:
# Print dataset info to see the column data types and check for null values
print(data.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   object 
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(2), object(3)
memory usage: 73.3+ KB


In [10]:
# Get basic statistical summary of the dataset
print(data.describe())

age          bmi     children       charges
count  1338.000000  1338.000000  1338.000000   1338.000000
mean     39.207025    30.663397     1.094918  13270.422265
std      14.049960     6.098187     1.205493  12110.011237
min      18.000000    15.960000     0.000000   1121.873900
25%      27.000000    26.296250     0.000000   4740.287150
50%      39.000000    30.400000     1.000000   9382.033000
75%      51.000000    34.693750     2.000000  16639.912515
max      64.000000    53.130000     5.000000  63770.428010

### Find out the average age of the patients in the dataset. ###

In [74]:
# calculate the mean age
mean_age = data.age.mean()

# print the result
print(f'The average age of patients in the dataset is {mean_age.round()} years.')

The average age of patients in the dataset is 39.0 years.


### Analyze where a majority of the individuals are from.

In [75]:
# count the number of individuals from each region
region_counts = data.region.value_counts()

# print the result
print(f"The majority of individuals in the dataset are from the {region_counts.idxmax()} region with {region_counts.max()} people.")

The majority of individuals in the dataset are from the southeast region with 364 people.


The most region from people - is the `Southeast` with `364` people

### Look at the different costs between smokers vs. non-smokers.

In [76]:
# calculate the average insurance cost for smokers and non-smokers
smoker_avg_cost = data.loc[data['smoker'] == 'yes', 'charges'].mean()
non_smoker_avg_cost = data.loc[data['smoker'] == 'no', 'charges'].mean()

# print the result
print(f"The average insurance cost for smokers is {smoker_avg_cost.round()}, which is more than {int(smoker_avg_cost // non_smoker_avg_cost)} times higher than the average insurance cost for non-smokers, which is {non_smoker_avg_cost.round()}.")


The average insurance cost for smokers is 32050.0, which is more than 3 times higher than the average insurance cost for non-smokers, which is 8434.0.


### Figure out what the average age is for someone who has at least one child in this dataset.

In [77]:
# calculate the average age for individuals with at least one child and those with no children
avg_age_with_child = data.loc[data['children'] > 0, 'age'].mean()
avg_age_without_child = data.loc[data['children'] == 0, 'age'].mean()

# print the result
print(f"The average age of individuals who have at least one child is {avg_age_with_child.round(1)} years old, compared to {avg_age_without_child.round(1)} years old for those with no children.")


The average age of individuals who have at least one child is 39.8 years old, compared to 38.4 years old for those with no children.


### How does BMI (body mass index) differ between smokers and non-smokers in the dataset?

In [79]:
# calculate the average BMI for smokers and non-smokers
smoker_avg_bmi = data.loc[data['smoker'] == 'yes', 'bmi'].mean()
non_smoker_avg_bmi = data.loc[data['smoker'] == 'no', 'bmi'].mean()

# print the result
print(f"The average BMI for smokers is {smoker_avg_bmi}, compared to {non_smoker_avg_bmi} for non-smokers. There is no significant difference.")


The average BMI for smokers is 30.70844890510949, compared to 30.651795112781954 for non-smokers. There is no significant difference.


### What is the average insurance charge for males versus females in the dataset?

In [80]:
# Calculate the average insurance charge for males and females separately
male_avg_cost = data.loc[data['sex'] == 'male', 'charges'].mean()
female_avg_cost = data.loc[data['sex'] == 'female', 'charges'].mean()

# Print the results with a formatted string
print(f"Average insurance charge for males: {male_avg_cost:.2f}")
print(f"Average insurance charge for females: {female_avg_cost:.2f}")

Average insurance charge for males: 13956.75
Average insurance charge for females: 12569.58
